solr/test-framework/src/java/org/apache/solr/cloud/MiniSolrCloudCluster.java

```diff
@@ -745,6 +753,30 @@ public CloudSolrClient getSolrClient() {
     return solrClient;
   }
 
+  public CloudSolrClient getSolrClientForCollection(String collectionName) {
```

Comment:
There is commonality here in what you are doing in this PR with the new Solr TestRule, as seen by this method.  In the TestRule, the "ForCollection" part is omitted for brevity.  WDYT?  Also needs javadocs to tell the caller not to close it; it's closed when the cluster is shutdown.